subprojects/version-info/version-info.gradle

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+task buildReceiptResource(type: Copy, dependsOn: rootProject.createBuildReceipt) {
+    into "$generatedResourcesDir/org/gradle"
+    from rootProject.createBuildReceipt.receiptFile
+}
+
+configurations {
+    testVersionInfo
+}
+
+sourceSets.main.output.dir generatedResourcesDir, builtBy: buildReceiptResource
+
+task createTestBuildReceipt(type: org.gradle.build.BuildReceipt) {
+    receiptFile = file("$buildDir/test-version-info-resources/org/gradle/build-receipt.properties")
+    versionNumber = rootProject.isSnapshot ? "${rootProject.versionBase}-snapshot-1" : rootProject.version
+    versionBase = rootProject.versionBase
+    snapshot = rootProject.isSnapshot
+}
+
+def libsDir = convention.getPlugin(BasePluginConvention).libsDir
+jar {
+    destinationDir = file("${libsDir}/prod")
+}
+
+task testVersionInfoJar(type: Jar, dependsOn: createTestBuildReceipt) {
```

Comment:
This would be more readable to me if `dependsOn` was part of the configuration closure, and not another parameter to the method call. We almost never use the latter in the Gradle build itself, and it's easy to overlook it.